src/app/helpers/advancedTypes.ts

```diff
@@ -34,3 +34,6 @@ export type KeysOfType<T, TProp> = {
 
 /** Sets type to either be T or null */
 export type Option<T> = T | null;
+
+/** Removed readonly flag from all properties in an object */
+export type Writeable<T> = {-readonly [P in keyof T]: T[P]}
```

Comment:
Highlight that this only affects the types, and does not change a read-only property to a writeable one

Reply:
Also, readonly in the model, may not be the same as readonly in the API